proposals/distribution.md

```diff
@@ -57,10 +58,92 @@ The API spec is currently considered v2 and we will start the specification at v
+        There is no need for the distribution specification to choose a particular authentication scheme, because clients receiving 401 and 407 responses can use IANA's [HTTP Authentication Scheme Registry][iana-auth] to look up referenced schemes and take appropriate action.
+        It is reasonable to provide a standardized way to use DNS based distribution in conjunction with OCI without requiring its use.
+
+* “Creating a reference spec for optional DNS based naming and discovery”.
+    Discovery and registry protocols are completely separate and do not need to be added together.
+    This entry replaces part of the previous “Creating Reference spec for optional DNS based naming & distribution” entry.
+
+    * What: Creating a reference spec for optional DNS based naming and discovery
+    * In/Out/Future: In scope for future specification
+    * Status: Not currently being worked
+    * Description: Define a protocol for resolving an image name to retrieval information.
+        When we address this, we will also allow for alternative, parallel name-to-image discovery protocols in parallel with the OCI-specified protocol.
+    * Why: It is reasonable to provide a standardized way to use DNS based distribution in conjunction with OCI without requiring its use.
+        There are many good use cases for DNS based distribution, but not all use cases support this.
+        Furthermore, encoding the location of a bundle into the bundle can cause issues with downloads from alternate locations other than the origin specified in the name.
+
+* “Specifying a distribution method”.
+    This entry replaces part of the previous “Creating Reference spec for optional DNS based naming & distribution” and “Standardizing on a particular Distribution method” entries.
+
+    Retrieving images covers the current “tag listing” (e.g. “what named manifests are in `library/busybox`?”), because tags are entries in the image format's [`manifests` array][manifests].
+    Other tag-listing endpoints needed for backwards-compatibility are therefore in scope as well.
+
+    Repository actions and listing images within a repository are considered part of distribution policy or content management and are out of scope for this entry's per-image action.
+    For example, “what images are under `library/`?” is out of scope for this project.
+
+    * What: Specifying a distribution method
+    * In/Out/Future: In scope
+    * Status: In progress (see opencontainers/distribution-spec)
+    * Description: Define a protocol for image, manifest, config, and blob creation, retrieval, and deletion.
+        Listing, creating, and deleting repositories and listing images within repositories are both out of scope for this entry.
+    * Why: This specification will provide one (of possibly many) distribution specifications.
+        Alternative distribution specifications may be developed for uses cases not covered by this specification, but defining them is out of scope for the OCI.
```

Comment:
I think it can be in-scope, probably after matured in the wild

Reply:
> I think it can be in-scope, probably after matured in the wild

You expect alternative, parallel distribution specifications to also move under the OCI umbrella?

Reply:
Potentially yes in future. But not right now.

So I suggest putting `... but defining them is out of scope for the OCI at this moment` to this document.

Reply:
> Potentially yes in future. But not right now.

I've pushed cdfde9b → fa29c25 with “is out of scope” to “is currently out of scope”, but if the plan is to hold this door open, I think we need to think this through more carefully.  Will the current docker/distribution API spec remain the “[Docker Registry HTTP API][1]” under the OCI?  Or will it become the “OCI Registry HTTP API”?  If it changes names, what will future additional OCI distribution specs be called?  Will they be maintained in the distribution-spec project created by #35, or will they get their own OCI Project?  If the plan is to use separate projects, maybe #35 should be adding a docker-registry-http-spec project/repository?  Or a registry-http-spec project/repository?

[1]: https://github.com/docker/distribution/blame/5cb406d511b7b9163bff9b6439072e4892e5ae3b/docs/spec/api.md#L7